torch/_functorch/aot_autograd.py

```diff
@@ -3698,6 +3824,14 @@ def _compiled_autograd_key(ctx):
         @staticmethod
         def forward(ctx, *deduped_flat_tensor_args):
             args = deduped_flat_tensor_args
+
+            marked_dirty_inps = []
+            if fw_metadata.keep_input_mutations and fw_metadata.is_train:
+                for ix, i in enumerate(deduped_flat_tensor_args):
```

Comment:
Instead of looping over all `deduped_flat_tensor_args`, can you loop over a list of indices of inputs that need to be marked dirty?

It's pretty common for a compiled function to have e.g. 200+ inputs, where only a handful are mutated, so we want to avoid looping over all the inputs in the hot-path (this function is part of the AOTAutograd runtime hotpath).